src/components/EmployeeDetail/EmployeeDetail.jsx

```diff
@@ -0,0 +1,209 @@
+  };
+
+  const returnRedirect = () => {
+    if (redirect) {
+      if (params === 'add') {
+        return (
+          <Redirect to="/employees"/>
+        );
+      } else {
+        const route = `/employee/${params}`;
+        return (
+          <Redirect to={route}/>
+        );
+      }
+    }
+  };
+
+  const onCancel = () => {
+    setEditClicked(false);
+  };
+
+  const onEditClick = () => {
+    setEditClicked(true);
+  };
+
+  const getRedirectHtml = () => {
+    if (params === 'add' || editClicked) {
+      const action = params === 'add' ? 'add' : 'edit';
+      return (
+        <React.Fragment>
+          {returnRedirect()}
+          <EmployeeEdit action={action}
+                        employee={employee}
+                        employeeId={params}
+                        handleCancel={onCancel}
+                        handleSave={saveEmployee}/>
+        </React.Fragment>
+      );
+    } else if (employee) {
+      return (
+        <div className={classes.root}>
+          <Paper className={classes.paper}>
+            <div className={classes.titleLink}>
+              <Link to='/employees' className={classes.link}>Back to list</Link>
+            </div>
+            <div className={classes.title}>
+              Employee Details
+            </div>
+            <div className={classes.wrapper}>
+              <div className={classes.container}>
+                <div className={classes.detailRow}>
+                  <div className={classes.label}>First Name</div>
+                  <div className={classes.input}>
+                    {employee.FirstName}
+                  </div>
+                </div>
+                <div className={classes.detailRow}>
+                  <div className={classes.label}>Last Name</div>
+                  <div className={classes.input}>
+                    {employee.LastName}
+                  </div>
+                </div>
+                <div className={classes.detailRow}>
+                  <div className={classes.label}>Title</div>
+                  <div className={classes.input}>
+                    {employee.Title}
+                  </div>
+                </div>
+                <div className={classes.detailRow}>
+                  <div className={classes.label}>Company</div>
+                  <div className={classes.input}>
+                    {employee.Company}
+                  </div>
+                </div>
+                <Divider/>
+                <div className={classes.formGroupButton}>
+                  <Button variant="contained"
+                          color="primary"
+                          className={classes.button}
+                          onClick={onEditClick}
+                          margin="normal">
+                    Edit
+                  </Button>
+                </div>
+              </div>
+            </div>
+          </Paper>
+        </div>
+      );
+    }
+  };
+
+  return (
+    <React.Fragment>
+      {getRedirectHtml()}
```

Comment:
Why would we render the content by this function call and not directly?

Reply:
Because of conditional rendering, I have used this function call.
We can also do inline conditional rendering but that would make the code clumsy and unreadable.